play.py

```diff
@@ -145,82 +145,86 @@ def play_aidungeon_2():
-
-            elif action.lower() == "help":
-                console_print(instructions())
-
-            elif action.lower() == "censor off":
-                if not generator.censor:
-                    console_print("Censor is already disabled.")
-                else:
-                    generator.censor = False
-                    console_print("Censor is now disabled.")
+            action = input("> ").strip()
+            if action[0] == "/":
+                split = action[1:].split(" ") # removes preceding slash
+                command = split[0].lower()
+                args = split[1:]
+                if command == "restart":
+                    rating = input("Please rate the story quality from 1-10: ")
+                    rating_float = float(rating)
+                    story_manager.story.rating = rating_float
+                    break
 
-            elif action.lower() == "censor on":
-                if generator.censor:
-                    console_print("Censor is already enabled.")
-                else:
-                    generator.censor = True
-                    console_print("Censor is now enabled.")
+                elif command == "quit":
+                    rating = input("Please rate the story quality from 1-10: ")
+                    rating_float = float(rating)
+                    story_manager.story.rating = rating_float
+                    exit()
+
+                elif command == "nosaving":
+                    upload_story = False
+                    story_manager.story.upload_story = False
+                    console_print("Saving turned off.")
+
+                elif command == "help":
+                    console_print(instructions())
+
+                elif command == "censor":
+                    if args[0] == "off":
+                        if not generator.censor:
+                            console_print("Censor is already disabled.")
+                        else:
+                            generator.censor = False
+                            console_print("Censor is now disabled.")
+
+                    elif args[0] == "on":
+                        if generator.censor:
+                            console_print("Censor is already enabled.")
+                        else:
+                            generator.censor = True
+                            console_print("Censor is now enabled.")
 
-            elif action.lower() == "save":
-                if upload_story:
-                    id = story_manager.story.save_to_storage()
-                    console_print("Game saved.")
-                    console_print(
-                        "To load the game, type 'load' and enter the following ID: "
-                        + id
-                    )
-                else:
-                    console_print("Saving has been turned off. Cannot save.")
+                    else:
+                        console_print(f"Invalid argument: {args[0]}")
```

Comment:
Is the minimum Python version documented anywhere? f-strings require Python 3.6, which shouldn't conflict with the Colab runtime, but may conflict with users running AIDungeon locally.

Reply:
IIRC, the install script specifically checks for python 3.6

Reply:
Sorry, completely missed that. Thanks!